### PR TITLE
add config section keys, improve verify_client documentation

### PR DESCRIPTION
**Pull request by tpo**
Sections in the yaml configuration file are organized under a section key. F.ex. the REST API parameters get configured under a `restapi` yaml key.

Those section keys are not mentioned in the SETTINGS documentation. This pull request fixes that. If adds the section key to each configuration section that has one (and indents all the settings accordingly to fall under that key).

Please pull. Thanks a lot for Patroni!
*t

### CODE REVIEW

**Comment by tpo** [comment]
@CyberDem0n  Finally I could also document the default value of `verify_client` - which one is it? `none`?

**Comment by CyberDem0n** [comment]
Yes, the default is `none`, i.e. client certificate isn't required.

File: docs/SETTINGS.rst [hunk 1]
```diff
@@ -147,83 +148,84 @@ Kubernetes
...
+        -  **keyfile**: (optional): Specifies the file with the secret key in the PEM format.
+        -  **cafile**: (optional): Specifies the file with the CA_BUNDLE with certificates of trusted CAs to use while verifying client certs.
+        -  **verify\_client**: (optional): ``none`` (default), ``optional`` or ``required``. When ``none`` REST API will not check client certificates. When ``required`` client certificates are required for all REST API calls. When ``optional`` client certificates are required for all unsafe REST API endpoints. If ``verify_client`` is set to ``optional`` or ``required`` basic-auth is not checked. When ``required`` is used, then client authentication succeeds, if the certificate signature verification succeeds.  For ``optional`` the client cert will only be checked for ``PUT``, ``POST``, ``PATCH``, and ``DELETE`` requests.
 
 .. _patronictl_settings:
 
 CTL
 ---
--  **Optional**:
+- **ctl**: (optional):
```
Comment (CyberDem0n) [inline docs/SETTINGS.rst:228]:
```suggestion
- **ctl**: (optional)
```

File: docs/SETTINGS.rst [hunk 2]
```diff
@@ -147,83 +148,84 @@ Kubernetes
...
+        
+        -  **authentication**: (optional):
             -  **username**: Basic-auth username to protect unsafe REST API endpoints.
             -  **password**: Basic-auth password to protect unsafe REST API endpoints.
-
-        -  **certfile**: Specifies the file with the certificate in the PEM format. If the certfile is not specified or is left empty, the API server will work without SSL.
-        -  **keyfile**: Specifies the file with the secret key in the PEM format.
-        -  **cafile**: Specifies the file with the CA_BUNDLE with certificates of trusted CAs to use while verifying client certs.
-        -  **verify\_client**: ``none``, ``optional`` or ``required``. When ``none`` REST API will not check client certificates. When ``required`` client certificates are required for all REST API calls. When ``optional`` client certificates are required for all unsafe REST API endpoints. If ``verify_client`` is set to ``optional`` or ``required`` basic-auth is not checked.
+
```
Comment (CyberDem0n) [inline docs/SETTINGS.rst:218]:
```suggestion

```

File: docs/SETTINGS.rst [hunk 3]
```diff
@@ -147,83 +148,84 @@ Kubernetes
...
+        
+        -  **listen**: IP address (or hostname) and port that Patroni will listen to for the REST API - to provide also the same health checks and cluster messaging between the participating nodes, as described above. to provide health-check information for HAProxy (or any other load balancer capable of doing a HTTP "OPTION" or "GET" checks).
+        
+        -  **authentication**: (optional):
```
Comment (CyberDem0n) [inline docs/SETTINGS.rst:215]:
```suggestion

        -  **authentication**: (optional)
```

**Comment by Jan-M** [comment]
:+1:

**Comment by CyberDem0n** [comment]
:+1:

**Comment by CyberDem0n** [comment]
Thank you @tpo!

**Comment by tpo** [comment]
Thank you @CyberDem0n !

**Pull request merged**